### PR TITLE
feat(taiko-client): improve progress tracker

### LINKED ISSUE #17160: feat(taiko-client): stop triggering another round of beacon sync when the node is still syncing

### PR DESCRIPTION
**Pull request by davidtaikocha**

### CODE REVIEW

File: packages/taiko-client/driver/chain_syncer/beaconsync/progress_tracker.go [hunk 1]
```diff
@@ -162,16 +163,34 @@ func (t *SyncProgressTracker) ClearMeta() {
...
 	defer t.mutex.RUnlock()
 
+	// If the beacon sync has not been triggered yet, we will simply trigger it.
 	if !t.triggered {
 		return true
 	}
 
-	return t.lastSyncedBlockID != nil && t.lastSyncedBlockID != newID
+	if t.lastSyncedBlockID == nil {
+		return false
```
Comment (mask-pp) [inline packages/taiko-client/driver/chain_syncer/beaconsync/progress_tracker.go:180]:
```suggestion
		return true
```

**Comment by mask-pp** [comment]
Actually `lastSyncedBlockID` is used for recording the current syncing `blockID`, how about rename to `lastSyncingBlockID`?

File: packages/taiko-client/driver/chain_syncer/beaconsync/progress_tracker.go (hunk 1 above)
Comment (YoGhurt111) [reply packages/taiko-client/driver/chain_syncer/beaconsync/progress_tracker.go:180]: Yep, it should be `true`

**Review by YoGhurt111 — APPROVED** [review]

File: packages/taiko-client/driver/chain_syncer/beaconsync/progress_tracker.go (hunk 1 above)
Comment (davidtaikocha) [reply packages/taiko-client/driver/chain_syncer/beaconsync/progress_tracker.go]: changed

**Review by mask-pp — APPROVED** [review]

**Pull request merged**